Indicators/StochasticRelativeStrengthIndex.cs

```diff
@@ -0,0 +1,128 @@
+            : this($"StochRSI({rsiPeriod},{stochPeriod},{kSmoothingPeriod},{dSmoothingPeriod},{movingAverageType})", rsiPeriod, stochPeriod, kSmoothingPeriod, dSmoothingPeriod, movingAverageType)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the StochasticRelativeStrengthIndex class
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="rsiPeriod">The period of the relative strength index</param>
+        /// <param name="stochPeriod">The period of the stochastic indicator</param>
+        /// <param name="kSmoothingPeriod">The smoothing period of K output</param>
+        /// <param name="dSmoothingPeriod">The smoothing period of D output</param>
+        /// <param name="movingAverageType">The type of moving average to be used</param>
+        public StochasticRelativeStrengthIndex(string name, int rsiPeriod, int stochPeriod, int kSmoothingPeriod, int dSmoothingPeriod, MovingAverageType movingAverageType = MovingAverageType.Simple)
+            : base(name)
+        {
+            _rsi = new RelativeStrengthIndex(rsiPeriod);
+            _recentRSIValues = new RollingWindow<decimal>(stochPeriod);
+
+            K = movingAverageType.AsIndicator($"{name}_K_{movingAverageType}", kSmoothingPeriod);
+            D = movingAverageType.AsIndicator($"{name}_D_{movingAverageType}", dSmoothingPeriod);
+
+            WarmUpPeriod = stochPeriod;
+        }
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => Samples >= WarmUpPeriod;
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod { get; }
+
+        /// <summary>
+        /// Computes the next value of the following sub-indicators from the given state:
+        /// K (%K) and D (%D)
+        /// </summary>
+        /// <param name="input">The input given to the indicator</param>
+        /// <returns>The input is returned unmodified.</returns>
+        protected override decimal ComputeNextValue(IndicatorDataPoint input)
+        {
+            _rsi.Update(input);
+
+            if (!IsReady)
+            {
+                return 0m;
+            }
+
+            _recentRSIValues.Add(_rsi.Current.Value);
+
+            var max_high = _recentRSIValues.Max();
+            var min_low = _recentRSIValues.Min();
+
+            decimal k = 100;
+            if (max_high != min_low)
+                k = 100 * (_rsi.Current.Value - min_low) / (max_high - min_low);
+
+            K.Update(input.EndTime, k);
+            D.Update(input.EndTime, K.Current.Value);
+
+            return input.Value;
```

Comment:
This can't be right, returning input value as the indicator value? Should expand the tests so they assert all indicator values K/D/Current 👍

Reply:
This look quite similar to what BB code is

https://github.com/QuantConnect/Lean/blob/3bdde74096904c9c4a6371a00a2308c1f37958ee/Indicators/BollingerBands.cs#L122-L129

Reply:
> This look quite similar to what BB code is

On interesting, I see how could it make sense, given the indicators don't have a clear single value, following the pattern sounds good to me 👍 can ignore comment

> ) {

Please move bracket into new line, following code style

```
            K.Update(input.EndTime, k);
            D.Update(input.EndTime, K.Current.Value);
```

Minor but curious why use `k` & `K.Current.Value` and not the same for both?

Reply:
k is current value while K.Current.Value is smoothed value of k.

talipp named K smoothed_k but return a class named StochRSIVal with a field named k.